cmd/gnodev/util.go

```diff
@@ -114,6 +115,28 @@ func guessRootDir() string {
 	return rootDir
 }
 
+func guessGnoModDir(absPath string) (string, error) {
+	if !filepath.IsAbs(absPath) {
+		return "", errors.New("requires absolute path")
+	}
+
+	for {
+		filePath := filepath.Join(absPath, "gno.mod")
+		_, err := os.Stat(filePath)
+		if err == nil {
+			return filePath, nil
+		}
+
+		absPath = filepath.Dir(absPath)
+		// Stop iterating if we have reached the root directory
+		if absPath == "/" {
```

Comment:
```suggestion
		if absPath == previousAbsPath {
```

For windows or other strange cases.